scripts/release/asset_publish.py

```diff
@@ -277,33 +279,60 @@ def _str2bool(v: str) -> bool:
             )
 
             # Handle specific asset types
-            if asset.type in [assets.AssetType.COMPONENT, assets.AssetType.ENVIRONMENT]:
-                # Assemble command
-                cmd = assemble_command(
-                    asset.type.value, str(asset.spec_with_path),
-                    registry_name, final_version, resource_group, workspace, debug_mode)
-                # Run command
-                run_command(cmd, failure_list, debug_mode)
+            if asset.type == assets.AssetType.COMPONENT:
+                # load component and check if environment exists
+                logger.print(f"spec's path: {asset.spec_with_path}")
+                component = load_component(asset.spec_with_path)
+                env = component.environment
+                ws_env_pattern = re.compile("^(.+):(.+)$")
+                registry_env_pattern = re.compile("^azureml://registries/.+/environments/(.+)/versions/(.+)")
```

Comment:
`ws_env_pattern` is a superset of `registry_env_pattern`. 

For example: `azureml://registries/r1/environments/e1/versions/1` matches both the regexes.

Reply:
I assume you are trying to distinguish b/w workspace vs registry based assets?

Reply:
Thank you. I have reversed the flow now. Checking registry env pattern first followed by ws env. Its just to fetch env name from string and then checking if env is already registered in the registry.